src/NUnitFramework/framework/Internal/Commands/OneTimeSetUpCommand.cs

```diff
@@ -75,15 +77,18 @@ public override TestResult Execute(TestExecutionContext context)
                     Test.Fixture = _suite.Fixture;
                 }
 
-                for (int i = _setUpTearDown.Count; i > 0; )
-                    _setUpTearDown[--i].RunSetUp(context);
+                if (!(suiteResult.Test is TestFixture && !suiteResult.Test.HasChildren))
```

Comment:
That does the job, but it does it by having one component (OneTimeSetUpCommand) be aware of the necessary behavior for different types of Suites. See my comments about how this all came about in #1962. I think we want a solution that makes individual suite types responsible for how their result is calculated based on the results of contained tests.